src/runtime-rs/crates/service/src/manager.rs

```diff
@@ -159,39 +159,20 @@ impl ServiceManager {
         Ok(())
     }
 
-    async fn send_event(&self, event: Arc<dyn Event>) -> Result<()> {
-        let any = Any {
-            type_url: event.type_url(),
-            value: event.value().context("get event value")?,
-            ..Default::default()
+    async fn new_event_publisher(namespace: &str) -> Result<Box<dyn Forwarder>> {
+        let fwd: Box<dyn Forwarder> = match env::var(TTRPC_ADDRESS_ENV) {
+            Ok(address) => Box::new(
+                ContainerdForwarder::new(namespace, &address)
+                    .await
+                    .context("new containerd forwarder")?,
+            ),
+            Err(_) => Box::new(
+                LogForwarder::new(namespace)
```

Comment:
I want to discuss if there is no `TTRPC_ADDRESS_ENV` should we use log or rollback to use the containerd binary to send the event?

Reply:
I think it's unnecessary to do fallback. The reasons are:

1. [Containerd doc](https://github.com/containerd/containerd/blob/main/runtime/v2/README.md#start) implies that this env will be available regardless of its version.
2. Runtime written in Go don't have that fallback either.